src/measure_theory/measure/finite_measure_weak_convergence.lean

```diff
@@ -0,0 +1,276 @@
+/-
+Copyright (c) 2021 Kalle Kytölä. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kalle Kytölä
+-/
+import measure_theory.measure.measure_space
+
+/-!
+# Weak convergence of (finite) measures
+
+This file will define the topology of weak convergence of finite measures and probability measures
+on topological spaces. The topology of weak convergence is the coarsest topology w.r.t. which
+for every bounded continuous `ℝ≥0`-valued function `f` on, the integration of `f` against the
+measure is continuous.
+
+TODOs:
+* Rename the types `finite_measures α` -> `finite_measure α` and
+  `probability_measures α` -> `probability_measure α` after PR #8831.
+* Define the topologies (the current version only defines the types) via
+  `weak_dual ℝ≥0 (α →ᵇ ℝ≥0)`.
+* Prove that an equivalent definition of the topologies is obtained using bounded continuous
+  `ℝ`-valued functions instead.
+* Include the portmanteau theorem on characterizations of weak convergence of (Borel) probability
+  measures.
+
+## Main definitions
+
+The main definitions are the types `finite_measures α` and `probability_measures α`.
+
+TODO:
+* Define the topologies on the above types.
+
+## Main results
+
+None yet.
+
+TODO:
+* Portmanteau theorem.
+
+## Notations
+
+No new notation is introduced.
+
+## Implementation notes
+
+The topology of weak convergence of finite Borel measures will be defined using a mapping from
+`finite_measures α` to `weak_dual ℝ≥0 (α →ᵇ ℝ≥0)`, inheriting the topology from there.
+
+## References
+
+* [Billingsley, *Convergence of probability measures*][billingsley1999]
+
+## Tags
+
+weak convergence of measures, finite measure, probability measure
+
+-/
+
+noncomputable theory
+open measure_theory
+open set
+open filter
+open_locale topological_space ennreal nnreal
+
+section finite_measures
+/-!
+### Finite measures
+In this section, we define the `Type` of finite measures on a given measurable space. This type
+will be equipped with the topology of weak convergence of measures when the measurable space is
+a topological space equipped with its Borel sigma-algebra.
+-/
+
+variables {α : Type} [measurable_space α]
+
+/-- Finite measures are defined as the subtype of measures that have the property of being finite
+measures (i.e., their total mass is finite). -/
+def finite_measures (α : Type*) [measurable_space α] : Type :=
+{ μ : measure α // finite_measure μ }
+
+namespace finite_measures
+
+instance has_zero {α : Type*} [measurable_space α] :
+  has_zero (finite_measures α) :=
+⟨{ val := 0,
+   property := measure_theory.finite_measure_zero, }⟩
+
+instance : inhabited (finite_measures α) := { default := 0 }
+
+instance {α : Type*} [measurable_space α] : has_add (finite_measures α) :=
+{ add := (λ (μ ν : finite_measures α),
+  { val := μ.val + ν.val,
+    property := @measure_theory.finite_measure_add α _ μ.val ν.val μ.prop ν.prop, }), }
```

Comment:
If you move this a little later, to after you set up the coercion and the `finite_measure` instance on the coercion, then Lean will figure out more things without you providing them using @ and implicit arguments:
```suggestion
  { val := μ + ν,
    property := measure_theory.finite_measure_add, }), }
```
(of course, I would also propose getting rid of the `val`/`property` here, as I mentioned before).

Same for the `has_scalar` instance.

Reply:
:bulb: Thank you!